### PR TITLE
chainparams: Add default fixed seeds for signet

### PR DESCRIPTION
**Pull request by maflcko**
This adds some random fixed seeds for signet. They might not work, but it is better than nothing.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Can be tested as specified in #20637 with an empty datadir: 

```

... (init)

# reading vSeeds:

2020-12-14T09:33:36Z trying connection ntv3mtqw5wt63red.onion:38333 lastseen=0.0hrs
2020-12-14T09:33:36Z Cannot create socket for ntv3mtqw5wt63red.onion:38333: unsupported network
2020-12-14T09:33:37Z trying connection 178.128.221.177 lastseen=0.0hrs
...
2020-12-14T09:33:38Z received: sendaddrv2 (0 bytes) peer=0
2020-12-14T09:33:38Z disconnecting peer=0


... (wait)

# reading vFixedSeeds:

2020-12-14T09:34:37Z Adding fixed seed nodes as DNS doesn't seem to be available.
2020-12-14T09:34:37Z IP 165.232.44.123 mapped to AS0 belongs to new bucket 527
2020-12-14T09:34:37Z IP 106.157.214.199 mapped to AS0 belongs to new bucket 668
2020-12-14T09:34:37Z IP 31.14.40.19 mapped to AS0 belongs to new bucket 313
2020-12-14T09:34:37Z IP 134.209.26.10 mapped to AS0 belongs to new bucket 668
2020-12-14T09:34:37Z IP 178.128.221.177 mapped to AS0 belongs to new bucket 367
2020-12-14T09:34:37Z IP 103.16.128.63 mapped to AS0 belongs to new bucket 843
2020-12-14T09:34:37Z IP 153.126.143.201 mapped to AS0 belongs to new bucket 744
2020-12-14T09:34:37Z IP 192.241.163.142 mapped to AS0 belongs to new bucket 830
2020-12-14T09:34:37Z trying connection 153.126.143.201:38333 lastseen=292.6hrs
2020-12-14T09:34:37Z Added connection peer=1
2020-12-14T09:34:37Z sending version (103 bytes) peer=1
2020-12-14T09:34:37Z send version message: version 70016, blocks=0, us=[::]:0, peer=1

**Comment by jonatack** [comment]
Maybe for mainnet tor v3 as well for bootstrapping connections in 0.21.0

**Comment by laanwj** [comment]
Concept ACK

**Comment by MarcoFalke** [comment]
> Maybe for mainnet tor v3 as well for bootstrapping connections in 0.21.0

I'll leave that for a follow-up

**Comment by Saibato** [comment]
@jonatack 
> Maybe for mainnet tor v3 as well for bootstrapping connections in 0.21.0

tyi, long V3 addresses cant be fixed encode, in the old style manner, since ADDRv2 breaks the logic to hide them in ipv6,
So i suggest to add them in chainparams, same effect since Tor skips anyway the DNS logic.

i.e.
            vSeeds.emplace_back("signetsuperv3boostrapxxxxxxxxxxx.onion:38333"); // V3 onion addres

File: contrib/seeds/generate-seeds.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
 '''
 Script to generate list of seed nodes for chainparams.cpp.
 
-This script expects two text files in the directory that is passed as an
+This script expects text files in the directory that is passed as an
```
Comment (jonatack) [inline contrib/seeds/generate-seeds.py:8]: fa2e9d9573dacf nit, `s/text files that is passed as an argument:/text files that are passed as arguments:/`

**Review by jonatack — COMMENTED** [review]
Tested and this appears to work.

```
2020-12-14T15:45:25Z init message: Loading P2P addresses...
2020-12-14T15:45:25Z ERROR: DeserializeFileDB: Failed to open file /home/jon/projects/bitcoin/bitcoin/datadir/signet/peers.dat
2020-12-14T15:45:25Z Invalid or missing peers.dat; recreating
2020-12-14T15:45:25Z tor: Got service ID t5acztv2shwvbbyfh6xktlgmryfhsw4ukloo2svlkhllfuwbzahtxlad, advertising service t5acztv2shwvbbyfh6xktlgmryfhsw4ukloo2svlkhllfuwbzahtxlad.onion:38333
2020-12-14T15:45:25Z AddLocal(t5acztv2shwvbbyfh6xktlgmryfhsw4ukloo2svlkhllfuwbzahtxlad.onion:38333,4)
2020-12-14T15:45:25Z ERROR: DeserializeFileDB: Failed to open file /home/jon/projects/bitcoin/bitcoin/datadir/signet/anchors.dat
2020-12-14T15:45:25Z 0 block-relay-only anchors will be tried for connections.
2020-12-14T15:45:25Z init message: Starting network threads...
2020-12-14T15:45:25Z addcon thread start
2020-12-14T15:45:25Z init message: Done loading
2020-12-14T15:45:25Z opencon thread start
2020-12-14T15:45:25Z dnsseed thread start
2020-12-14T15:45:25Z net thread start
2020-12-14T15:45:25Z msghand thread start
2020-12-14T15:45:25Z Loading addresses from DNS seed 178.128.221.177
2020-12-14T15:45:25Z Loading addresses from DNS seed 2a01:7c8:d005:390::5
2020-12-14T15:45:25Z Loading addresses from DNS seed ntv3mtqw5wt63red.onion:38333
2020-12-14T15:45:25Z 0 addresses found from DNS seeds
2020-12-14T15:45:25Z dnsseed thread exit
2020-12-14T15:45:27Z New outbound peer connected: version: 70016, blocks=15997, peer=0 (full-relay)
2020-12-14T15:46:26Z Adding fixed seed nodes as DNS doesn't seem to be available.
2020-12-14T15:46:26Z New outbound peer connected: version: 70016, blocks=15997, peer=1 (full-relay)
2020-12-14T15:46:27Z Synchronizing blockheaders, height: 2000 (~12.21%)
```

File: contrib/seeds/generate-seeds.py (hunk 1 above)
Comment (MarcoFalke) [reply contrib/seeds/generate-seeds.py:8]: That'd be wrong. The directory is passed, not the text files.

Comment (jonatack) [reply contrib/seeds/generate-seeds.py:8]: Oh ok, I misunderstood on reading. Maybe: "expects to find text files in the directory that is passed as an argument:"

Comment (MarcoFalke) [reply contrib/seeds/generate-seeds.py:8]: Leaving doc rewording for a follow-up, which I am happy to review

**Comment by MarcoFalke** [comment]
Added to 0.21.1 milestone

**Review by kallewoof — APPROVED** [review]
utACK fae6558de419b5825f11168f068058ea442fe131

**Comment by laanwj** [comment]
There's no instructions in the release notes for updating this. I'm not sure this is necessary, but for example the testnet hardcoded seeds haven't been updated in ages, for example, and it's unclear how useful they still are.

**Comment by MarcoFalke** [comment]
The list was generated with `getpeerinfo | grep addr`. I am happy to add that to the release notes, but surely there are better ways to get a list in the future. Maybe someone will even run a dnsseeder.

**Comment by ajtowns** [comment]
Looks okay to me

**Comment by Sjors** [comment]
What problem is this solving? More signet seeds just in case?

**Comment by MarcoFalke** [comment]
> What problem is this solving? More signet seeds just in case?

fixed seeds are used as fallback seeds when the dns seeds can't be queried for some reason

**Comment by MarcoFalke** [comment]
Rebased to get rid of already merged hunks. No other changes.

File: contrib/seeds/nodes_signet.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# List of fixed seed nodes for signet
+
+165.232.44.123:38333
+106.157.214.199:38333
+31.14.40.19:38333
+134.209.26.10:38333
+178.128.221.177:38333
+103.16.128.63:38333
+153.126.143.201:38333
+192.241.163.142:38333
```
Comment (ajtowns) [inline contrib/seeds/nodes_signet.txt:10]: Only:

```
    "address": "178.128.221.177",
    "address": "103.16.128.63",
    "address": "153.126.143.201",
    "address": "192.241.163.142",
```

seem to still be present?

**Review by ajtowns — COMMENTED** [review]
utACK fae1a64a18a72a760c56fde2f6795d56a535911f

I suppose running a dns seed and getting statistics from there for populating nodes_signet.txt is the way to get good values here.

File: contrib/seeds/nodes_signet.txt (hunk 2 above)
Comment (MarcoFalke) [reply contrib/seeds/nodes_signet.txt:10]: I think without a dns seeder or other software to track uptime, there will always be nodes in the list that are not reachable

**Comment by laanwj** [comment]
Hadn't really noticed this before it conflicted with the TorV3 one. Why close this?

**Comment by MarcoFalke** [comment]
Didn't seem a lot of appetite and v3 for mainnet is more important than signet

**Pull request closed without merging**